web/src/app/users/UserCalendarSubscriptionList.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react'
+import { PropTypes as p } from 'prop-types'
+import { useQuery } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+import { Link } from 'react-router-dom'
+import { Card, Tooltip } from '@material-ui/core'
+import FlatList from '../lists/FlatList'
+import OtherActions from '../util/OtherActions'
+import CreateFAB from '../lists/CreateFAB'
+import CalendarSubscribeCreateDialog from '../schedules/calendar-subscribe/CalendarSubscribeCreateDialog'
+import { Warning } from '../icons'
+import CalendarSubscribeDeleteDialog from '../schedules/calendar-subscribe/CalendarSubscribeDeleteDialog'
+import CalendarSubscribeEditDialog from '../schedules/calendar-subscribe/CalendarSubscribeEditDialog'
+import { GenericError, ObjectNotFound } from '../error-pages'
+import _ from 'lodash-es'
+import Spinner from '../loading/components/Spinner'
+import { formatTimeSince } from '../util/timeFormat'
+
+export const calendarSubscriptionsQuery = gql`
+  query calendarSubscriptions($id: ID!) {
+    user(id: $id) {
+      id
+      calendarSubscriptions {
+        id
+        name
+        reminderMinutes
+        scheduleID
+        schedule {
+          name
+        }
+        lastAccess
+        disabled
+      }
+    }
+  }
+`
+
+export default function UserCalendarSubscriptionList(props) {
+  const [showCreateDialog, setShowCreateDialog] = useState(false)
+  const [showEditDialogByID, setShowEditDialogByID] = useState(null)
+  const [showDeleteDialogByID, setShowDeleteDialogByID] = useState(null)
+
+  const { data, loading, error } = useQuery(calendarSubscriptionsQuery, {
+    variables: {
+      id: props.userID,
+    },
+  })
+
+  if (error) return <GenericError error={error.message} />
+  if (!_.get(data, 'user.id')) return loading ? <Spinner /> : <ObjectNotFound />
+
+  // sort by schedule names, then subscription names
+  const subs = data.user.calendarSubscriptions.sort((a, b) => {
+    if (a.schedule.name < b.schedule.name) return -1
+    if (a.schedule.name > b.schedule.name) return 1
+
+    if (a.name > b.name) return 1
+    if (a.name < b.name) return -1
+  })
+
+  const subheaderDict = {}
+  const items = []
+
+  // push schedule names as subheaders now that the array is sorted
+  subs.forEach(sub => {
+    if (!subheaderDict[sub.schedule.name]) {
+      subheaderDict[sub.schedule.name] = true
+      items.push({
+        subHeader: (
+          <Link to={`/schedules/${sub.scheduleID}`}>{sub.schedule.name}</Link>
```

Comment:
subHeader is expecting a string but receiving an element here, probably want to add element as an accepted type for subHeader in FlatList.